_posts/2019-10-07-vuejs-london-2019.md

```diff
@@ -0,0 +1,499 @@
+
+There are three kinds of routers possible:
+
+* Imperative: we define the routes and add a callback function to determine how the page should be rendered. An example of this is [page.js](https://github.com/visionmedia/page.js){:target="_blank" rel="noopener noreferrer"}.
+* Declarative: routes are defined on a component level by using for example a specific attribute. The [Reach Router](https://reach.tech/router/){:target="_blank" rel="noopener noreferrer"} is an example of this.
+* Configuration based: routes are defined separately with the components that they are linked to. The Vue router is a configuration based router.
+
+Because of this configuration based nature of the `vue-router`, we are missing the option to add or remove routes and we also lack declarative routing.
+
+The current implementation of `vue-router` has a lot of functionalities built in that are specific to managing the history while we actually want to separate this.
+In an ideal situation, we want the router to be the connection between the components and the code that manages the browser history.
+When a user navigates to a new page, the router calls the history code to add the new page to the history.
+When the user navigates back, the router retrieves the previous page from the history.
+In the current implementation, this is all not clearly separated.
+
+So what are the next steps for the Vue router?
+One major improvement would be the use of TypeScript as this would help developers on using and extending the Vue router in the right way.
+This is in the same spirit as Vue 3 that is also written fully in TypeScript.
+
+Another improvement lies in the use of a ranking system for the routes.
+Currently routes are checked for a match based on the order in which they are defined.
+This means that a catch-all route such as `(.*)` in the beginning of the routes definition would render the other routes useless.
+
+A major addition to Vue router would be the ability to have dynamic routing.
+Components could thus add or remove routes.
+A final addition would be the full use of the Composition API so we have functionalities such as `useLink`, `useLocation`, `onBeforeRouteLeave` and `onBeforeRouteUpdate`.
+
+The focus of the new router will be mainly on supporting Vue 3 while smaller improvements will be done for Vue 2.
+
+> [View slides](https://slides.com/posva/a-new-router-to-guide-your-apps){:target="_blank" rel="noopener noreferrer"}
+
+## Lightning talks
+
+### How to get your product owners to write your functional tests, by [Callum Silcock](https://twitter.com/csiilk){:target="_blank" rel="noopener noreferrer"}
+With the help of the [Cypress Cucumber preprocessor](https://github.com/TheBrainFamily/cypress-cucumber-preprocessor){:target="_blank" rel="noopener noreferrer"} Callum was able demonstrate how code like this could be processed by Cypress:
```

Comment:
typo: seems like a "to" is missing.
"was able demonstrate" -> "was able to demonstrate"

Reply:
Fixed, thx! :)